### PR TITLE
Check CORS Headers

### PR DESCRIPTION
**Pull request by beaufortfrancois**
R= @mounirlamouri 

screenshot 2016-01-13 at 1 45 03 pm

Fix https://github.com/mounirlamouri/manifest-validator/issues/27

### CODE REVIEW

**Comment by beaufortfrancois** [comment]
@mounirlamouri Please have a look.

**Comment by beaufortfrancois** [comment]
@mounirlamouri (gentle ping)

File: main.py [hunk 1]
```diff
@@ -30,14 +31,26 @@ def post(self):
           try:
             manifest_result = urlfetch.fetch(manifest_url)
             if manifest_result.status_code == 200:
+              manifest_url = manifest_result.final_url or manifest_url
               response['websiteUrl'] = website_url
               response['manifestUrl'] = manifest_url
-              response['content'] = manifest_result.content
-              # TODO Check CORS headers
+              # If same origin, return content.
+              if urlparse(manifest_url).netloc == urlparse(website_url).netloc:
```
Comment (mounirlamouri) [inline main.py]: so netloc is an origin?

File: main.py [hunk 2]
```diff
@@ -30,14 +31,26 @@ def post(self):
...
+                response['content'] = manifest_result.content
+              # Check for CORS Headers.
+              elif 'Access-Control-Allow-Origin' not in manifest_result.headers:
+                response['error'] = 'Access-Control-Allow-Origin HTTP header is not set.'
+              elif (manifest_result.headers['Access-Control-Allow-Origin'] != '*' or \
+                  website_url not in manifest_result.headers['Access-Control-Allow-Origin']):
```
Comment (mounirlamouri) [inline main.py]: shouldn't you check the origin too?

**Comment by mounirlamouri** [comment]
So sorry for taking so long to look at this. It lgtm with the comments addressed. Though, I wonder if it wouldn't be better for compat to actually grab the header info, send them back to the client and have the client do the checks. We now have an URL parser in the Web platform and it's more likely this parser will match the rules that the cors headers rules are using.

**Comment by beaufortfrancois** [comment]
The new URL parser is promising for sure.
I'll come up with another PR that return Header Info instead and do check in JS.

**Comment by beaufortfrancois** [comment]
I've reopened this PR and added JS side for CORS Header.
@mounirlamouri Let me know what you think.

File: static/ui.js [hunk 3]
```diff
@@ -77,10 +77,23 @@ document.querySelector('#check-website-url').onclick = function(e) {
...
+      }
     } else {
-      ManifestParser.parse(this.response.content);
-      _showParserLogs({manifestUrl : this.response.manifestUrl});
-      _showParserTips();
+      var manifestUrl = new URL(this.response.manifestUrl);
+      var websiteUrl = new URL(this.response.websiteUrl);
+      if (manifestUrl.origin == websiteUrl.origin ||
+          (this.response.originHeader &&
+          (this.response.originHeader.includes(websiteUrl.origin) ||
```
Comment (mounirlamouri) [inline static/ui.js]: nit: one space here to make it clear, this is part of the same condition as the line above?

File: static/ui.js [hunk 4]
```diff
@@ -77,10 +77,23 @@ document.querySelector('#check-website-url').onclick = function(e) {
...
+          this.response.originHeader == '*'))) {
```
Comment (mounirlamouri) [inline static/ui.js]: ditto

**Comment by mounirlamouri** [comment]
lgtm! :)

File: static/ui.js (hunk 3 above)
Comment (beaufortfrancois) [inline static/ui.js]: Done.

File: static/ui.js (hunk 4 above)
Comment (beaufortfrancois) [inline static/ui.js]: Done.

**Comment by beaufortfrancois** [comment]
Merged ;)

**Pull request merged**